### PR TITLE
Fetch connectors data from slack endpoint

### PR DESCRIPTION
**Pull request by WilliamBergamin**
## Summary

This PR aims to allow the generation script to fetch the connector apps information from `apps.certified.schema.list` endpoint.

This will create an easier generation process and will allow for automated code generation.

Testing: follow the generation steps in the `maintainers_guide`

Note: There are more changes then expected since some files were renamed, some tests were updated and I took this opportunity to improve the code.

## Requirements

- [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/deno-slack-sdk/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
- [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: .github/maintainers_guide.md [hunk 1]
```diff
@@ -43,23 +43,20 @@ file to import/export all of the defined functions.
...
+2. This will output something like the following:
 
    ```txt
    Cleaning folder directory
    ...
-   Generated 19 Connectors with their unit tests
-   Updated functions module export
    Formatting Slack function files...
    Linting Slack function files...
    ```
```
Comment (filmaj) [inline .github/maintainers_guide.md:62]: Should the reference to `functions.json` in the next paragraph below also be removed?

Comment (filmaj) [inline .github/maintainers_guide.md:62]: As part of releasing, is it expected to run the generation script? If so, maybe we should update the release steps below? I.e. run generation, commit the changes, push up, etc.

**Review by filmaj — APPROVED** [review]
Looks good to me! Worked well. I didn't review the code in-depth, more checked that things worked. If there's specific parts of the code, or approaches/methods, you want me to review, let me know.

Left a few comments mainly around documentation and releasing steps.

File: .github/maintainers_guide.md (hunk 1 above)
Comment (WilliamBergamin) [reply .github/maintainers_guide.md:62]: Yess good catch 💯

Comment (WilliamBergamin) [reply .github/maintainers_guide.md:62]: I think we should keep this separate for now, in the near future `run generation, commit the changes, push up, etc.` may be automated in a GitHub action and all the maintainers will have to do to "release" follow the current `Releasing steps`

**Pull request merged**